### PR TITLE
4-to-5 migration improvments

### PR DESCRIPTION
**Pull request by kevina**
Try to recover from an interrupted migration in `ipfs-4-to-5` and add a `-no-revert` option to avoid automatically reverting in case of failure.

This doesn't cover all failure cases but covers the most common ones (mainly flatfs.Move being interrupted or failing for some reason).  For other failure cases it should be possible to manually fix up the repo and then rerunning `ipfs-4-to-5`.

### CODE REVIEW

**Comment by kevina** [comment]
not sure how this will work,  I made the pull request on top of another pull request, as in it is based on ` feat/4-to-5 ` and not master

File: ipfs-4-to-5/migration/migration.go [hunk 1]
```diff
@@ -53,7 +53,16 @@ func (m Migration) Apply(opts migrate.Options) error {
 	basepath := filepath.Join(opts.Path, "blocks")
 	ffspath := filepath.Join(opts.Path, "blocks-v4")
 	if err := os.Rename(basepath, ffspath); err != nil {
-		return err
+		if os.IsNotExist(err) {
+			fi, err2 := os.Stat(basepath)
```
Comment (whyrusleeping) [inline ipfs-4-to-5/migration/migration.go]: don't you mean to stat ffspath here?

Comment (kevina) [inline ipfs-4-to-5/migration/migration.go]: Yep.

**Comment by kevina** [comment]
@whyrusleeping okay.  I am going to add a "--no-revert" option as part of the p.r. this will make failure and recovery easier to test.

**Comment by kevina** [comment]
@whyrusleeping assuming the tests pass this should be ready now

**Comment by kevina** [comment]
note that if you "merge" this it should go into #47 and not master ...

File: go-migrate/migrate.go [hunk 2]
```diff
@@ -29,6 +29,13 @@ type Migration interface {
 	Revert(Options) error
 }
 
+type SupportNoRevert interface {
```
Comment (whyrusleeping) [inline go-migrate/migrate.go]: This is weird... I would just make it part of the interface and backfill it

Comment (whyrusleeping) [inline go-migrate/migrate.go]: Or make it more general somehow. This is a very weird single purpose interface

Comment (kevina) [inline go-migrate/migrate.go]: @whyrusleeping I only did this becuase the readme says:

> Frozen. After the tool is written, all code must be frozen and vendored.

I took that that old versions of the conversion tool should not change.  I will happy to backfill it if you think it is better.

Comment (whyrusleeping) [inline go-migrate/migrate.go]: Ah, hrm...  Maybe then a simple array of booleans for checking which versions are supported would be the cleanest solution here

Comment (kevina) [inline go-migrate/migrate.go]: okay will do

**Pull request merged**